### ISSUE TITLE
FR: White background color and some guard space around barcode after composing

### ISSUE THREAD

**Issue by EyeCon** [markusfisch/BinaryEye#137]
Hello, I really like BinaryEye and appreciate the work you invested for maintaining it and listening to the users. Thank you!

I have one small feature request: I really like that we can compose barcodes and export them, but what would be better is if we could have the displayed barcode ready to be scanned, without exporting.

Currently the barcode has a dark background (at least on my phone, I don't know the color is decided) and is so close to the screen edge that it cannot be scanned reliably. Would it be possible that the background color be configurable (or white) when composing, and optionally have a way to scale down/up the created barcode size? Maybe with pinch gesture or having a spinbox for the scale factor?

**Comment by markusfisch** [comment]
Yes, that makes a lot of sense 👍  And I ran into this problem myself too 😉 

Unfortunately, ZXing doesn't add the required white space for all barcodes it can generate (although not all require a white space, e.g. a Aztec code). So the app should really add some extra margin. And I already thought about a button to make the background white too.

So expect that to be in the next version.

**Comment by VladWinner** [comment]
@markusfisch Is there any update on the issue? I noticed that some industrial QR scanners totally refuse to read the resulting white code on a black background, which makes it necessary to take extra steps, which is sometimes inconvenient
